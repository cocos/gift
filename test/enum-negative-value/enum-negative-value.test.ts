
import ps from 'path';
import fs from 'fs-extra';
import { bundle } from '../../source/gift';

describe('enum negative value export', () => {

    test('enum negative value', async () => {
        const inputPath = ps.join(__dirname, 'input.d.ts');
        const outputPath = ps.join(__dirname, 'output.d.ts');
        const result = bundle({
            input: inputPath,
            rootModule: 'index',
            name: 'out/index',
            output: outputPath,
        });
        expect(result.groups.length).toBe(1);
        expect(result.groups[0].code).toMatchSnapshot();
    });

});