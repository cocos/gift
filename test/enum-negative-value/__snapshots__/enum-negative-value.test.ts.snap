// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum negative value export enum negative value 1`] = `
"declare module \\"out/index\\" {
    export enum NegativeValue {
        AAA = \\"Bad expression <-1>\\",
        BBB = \\"Bad expression <-2>\\",
        CCC = 0,
        DDD = 1,
        EEE,
        FFF
    }
    export {};
}
"
`;
